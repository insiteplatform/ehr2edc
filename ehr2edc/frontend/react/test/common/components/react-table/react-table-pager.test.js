import React from 'react';
import renderer from 'react-test-renderer';
import ReactTablePager from "../../../../src/common/components/react-table/react-table-pager";

describe("Pager rendering", () => {
    test("Pager arguments get rendered well", () => {
        const pager = renderer.create(<ReactTablePager page={12} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]}/>);
        let tree = pager.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("A pager with a small amount of pages gets rendered well", () => {
        const pager = renderer.create(<ReactTablePager page={1} pageSize={10} pages={3} pageSizeOptions={[2, 5, 10]}/>);
        let tree = pager.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("First page selected gets rendered well", () => {
        const pager = renderer.create(<ReactTablePager page={0} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]}/>);
        let tree = pager.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Last page selected gets rendered well", () => {
        const pager = renderer.create(<ReactTablePager page={29} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]}/>);
        let tree = pager.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Pager change called gets rendered well", () => {
        let pageChangedEvent = jest.fn();
        let pager = renderer.create(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        const props = {
            page: 5,
            pageSize: 20,
            pages: 10,
            pageSizeOptions: [5, 20, 50]
        };
        pager.getInstance().componentWillReceiveProps(props);

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
    });
    test("Pager change called with the same amount of pages gets rendered well", () => {
        let pageChangedEvent = jest.fn();
        let pager = renderer.create(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        const props = {
            page: 5,
            pageSize: 30,
            pages: 30,
            pageSizeOptions: [5, 20, 50]
        };
        pager.getInstance().componentWillReceiveProps(props);

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
    });
});
describe("Page size navigation", () => {
    test("Changing the pager size triggers the event", () => {
        let pageSizeChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageSizeChange={pageSizeChangedEvent}/>);

        pager.find('.page-size-5 a').simulate('click');

        expect(pageSizeChangedEvent.mock.calls.length).toEqual(1);
        pager.unmount();
    });
});
describe("Page navigation", () => {
    test("Going to the first page triggers the event", () => {
        let pageChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        pager.find('.pager-navigation-button.first a').simulate('click');

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
        pager.unmount();
    });
    test("Going to the previous page triggers the event", () => {
        let pageChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        pager.find('.pager-navigation-button.prev a').simulate('click');

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
        pager.unmount();
    });
    test("Going to the next page triggers the event", () => {
        let pageChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        pager.find('.pager-navigation-button.next a').simulate('click');

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
        pager.unmount();
    });
    test("Going to the last page triggers the event", () => {
        let pageChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        pager.find('.pager-navigation-button.last a').simulate('click');

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
        pager.unmount();
    });
    test("Going to a certain page triggers the event", () => {
        let pageChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={30} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        pager.find('.pager-navigation-page.page-28 a').simulate('click');

        expect(pageChangedEvent.mock.calls.length).toEqual(1);
        pager.unmount();
    });
    test("Clicking the same page as you're on does not trigger the event", () => {
        let pageChangedEvent = jest.fn();
        let pager = mount(<ReactTablePager page={28} pageSize={10} pages={30} pageSizeOptions={[2, 5, 10]} onPageChange={pageChangedEvent}/>);

        pager.find('.pager-navigation-page.page-29 a').simulate('click');

        expect(pageChangedEvent.mock.calls.length).toEqual(0);
        pager.unmount();
    });
});