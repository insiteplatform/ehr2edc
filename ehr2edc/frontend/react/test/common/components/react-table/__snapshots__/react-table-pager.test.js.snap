// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager rendering A pager with a small amount of pages gets rendered well 1`] = `
<zf-list-footer>
  <div
    className="pager-size"
  >
    <span>
      Items per page: 
    </span>
    <ul>
      <li
        className="page-size-2"
      >
        <a
          onClick={[Function]}
        >
          2
        </a>
      </li>
      <li
        className="page-size-5"
      >
        <a
          onClick={[Function]}
        >
          5
        </a>
      </li>
      <li
        className="page-size-10 is-active"
      >
        <a
          onClick={[Function]}
        >
          10
        </a>
      </li>
    </ul>
  </div>
  <div
    className="pager-middle"
  />
  <ul
    className="pager"
  >
    <li
      className="pager-navigation-button first"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button prev"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-page page-1"
    >
      <a
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="pager-navigation-page is-active page-2"
    >
      <a
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="pager-navigation-page page-3"
    >
      <a
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="pager-navigation-button next"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-right"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button last"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-right"
        />
      </a>
    </li>
  </ul>
</zf-list-footer>
`;

exports[`Pager rendering First page selected gets rendered well 1`] = `
<zf-list-footer>
  <div
    className="pager-size"
  >
    <span>
      Items per page: 
    </span>
    <ul>
      <li
        className="page-size-2"
      >
        <a
          onClick={[Function]}
        >
          2
        </a>
      </li>
      <li
        className="page-size-5"
      >
        <a
          onClick={[Function]}
        >
          5
        </a>
      </li>
      <li
        className="page-size-10 is-active"
      >
        <a
          onClick={[Function]}
        >
          10
        </a>
      </li>
    </ul>
  </div>
  <div
    className="pager-middle"
  />
  <ul
    className="pager"
  >
    <li
      className="pager-navigation-button first disabled"
    >
      <a
        disabled={true}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button prev disabled"
    >
      <a
        disabled={true}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-page is-active page-1"
    >
      <a
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="pager-navigation-page page-2"
    >
      <a
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="pager-navigation-page page-3"
    >
      <a
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="pager-navigation-page page-4"
    >
      <a
        onClick={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="pager-navigation-page page-5"
    >
      <a
        onClick={[Function]}
      >
        5
      </a>
    </li>
    <a
      className="pager-navigation-page"
    >
      ...
    </a>
    <li
      className="pager-navigation-page page-30"
    >
      <a
        onClick={[Function]}
      >
        30
      </a>
    </li>
    <li
      className="pager-navigation-button next"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-right"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button last"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-right"
        />
      </a>
    </li>
  </ul>
</zf-list-footer>
`;

exports[`Pager rendering Last page selected gets rendered well 1`] = `
<zf-list-footer>
  <div
    className="pager-size"
  >
    <span>
      Items per page: 
    </span>
    <ul>
      <li
        className="page-size-2"
      >
        <a
          onClick={[Function]}
        >
          2
        </a>
      </li>
      <li
        className="page-size-5"
      >
        <a
          onClick={[Function]}
        >
          5
        </a>
      </li>
      <li
        className="page-size-10 is-active"
      >
        <a
          onClick={[Function]}
        >
          10
        </a>
      </li>
    </ul>
  </div>
  <div
    className="pager-middle"
  />
  <ul
    className="pager"
  >
    <li
      className="pager-navigation-button first"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button prev"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-page page-1"
    >
      <a
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <a
      className="pager-navigation-page"
    >
      ...
    </a>
    <li
      className="pager-navigation-page page-27"
    >
      <a
        onClick={[Function]}
      >
        27
      </a>
    </li>
    <li
      className="pager-navigation-page page-28"
    >
      <a
        onClick={[Function]}
      >
        28
      </a>
    </li>
    <li
      className="pager-navigation-page page-29"
    >
      <a
        onClick={[Function]}
      >
        29
      </a>
    </li>
    <li
      className="pager-navigation-page is-active page-30"
    >
      <a
        onClick={[Function]}
      >
        30
      </a>
    </li>
    <li
      className="pager-navigation-button next disabled"
    >
      <a
        disabled={true}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-right"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button last disabled"
    >
      <a
        disabled={true}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-right"
        />
      </a>
    </li>
  </ul>
</zf-list-footer>
`;

exports[`Pager rendering Pager arguments get rendered well 1`] = `
<zf-list-footer>
  <div
    className="pager-size"
  >
    <span>
      Items per page: 
    </span>
    <ul>
      <li
        className="page-size-2"
      >
        <a
          onClick={[Function]}
        >
          2
        </a>
      </li>
      <li
        className="page-size-5"
      >
        <a
          onClick={[Function]}
        >
          5
        </a>
      </li>
      <li
        className="page-size-10 is-active"
      >
        <a
          onClick={[Function]}
        >
          10
        </a>
      </li>
    </ul>
  </div>
  <div
    className="pager-middle"
  />
  <ul
    className="pager"
  >
    <li
      className="pager-navigation-button first"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button prev"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-left"
        />
      </a>
    </li>
    <li
      className="pager-navigation-page page-1"
    >
      <a
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <a
      className="pager-navigation-page"
    >
      ...
    </a>
    <li
      className="pager-navigation-page page-11"
    >
      <a
        onClick={[Function]}
      >
        11
      </a>
    </li>
    <li
      className="pager-navigation-page page-12"
    >
      <a
        onClick={[Function]}
      >
        12
      </a>
    </li>
    <li
      className="pager-navigation-page is-active page-13"
    >
      <a
        onClick={[Function]}
      >
        13
      </a>
    </li>
    <a
      className="pager-navigation-page"
    >
      ...
    </a>
    <li
      className="pager-navigation-page page-30"
    >
      <a
        onClick={[Function]}
      >
        30
      </a>
    </li>
    <li
      className="pager-navigation-button next"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-right"
        />
      </a>
    </li>
    <li
      className="pager-navigation-button last"
    >
      <a
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-angle-double-right"
        />
      </a>
    </li>
  </ul>
</zf-list-footer>
`;
